### PR TITLE
Add calculation and display of match %

### PR DESCRIPTION
**Pull request by ethteck**

### CODE REVIEW

File: frontend/src/components/ScoreBadge.tsx [hunk 1]
```diff
@@ -3,11 +3,24 @@ import classNames from "classnames"
 
 import styles from "./ScoreBadge.module.scss"
 
+function round(num: number) {
+    return Math.round((num + Number.EPSILON) * 100) / 100
```
Comment (nanaian) [inline frontend/src/components/ScoreBadge.tsx:7]: ELI5 pls? In a comment, perhaps?

**Review by nanaian — CHANGES_REQUESTED** [review]
Use camelCase for max_score (maxScore) for all but the `DiffOutput` type coming from the API

File: frontend/src/components/ScoreBadge.tsx (hunk 1 above)
Comment (mkst) [reply frontend/src/components/ScoreBadge.tsx:7]: `// force gud rounding https://stackoverflow.com/a/11832950/7290888`

Comment (nanaian) [reply frontend/src/components/ScoreBadge.tsx:7]: update: use toFIxed(2) to convert to string instead of rounding then converting that

File: frontend/src/components/ScoreBadge.tsx [hunk 2]
```diff
@@ -3,11 +3,20 @@ import classNames from "classnames"
 
 import styles from "./ScoreBadge.module.scss"
 
+function calculateScorePercent(score: number, maxScore: number) {
+    if (score > maxScore) {
+        return 0
```
Comment (nanaian) [inline frontend/src/components/ScoreBadge.tsx:8]:
```suggestion
function calculateScorePercent(score: number, maxScore: number): string {
    if (score > maxScore) {
        return "0"
```

**Review by nanaian — CHANGES_REQUESTED** [review]
types

File: frontend/src/components/ScoreBadge.tsx (hunk 2 above)
Comment (ethteck) [reply frontend/src/components/ScoreBadge.tsx:8]: I disagree that this should be returning a string

Comment (ethteck) [reply frontend/src/components/ScoreBadge.tsx]: overruled

**Review by nanaian — APPROVED** [review]
